cv_next/app/feed/components/CVItem.tsx

```diff
@@ -65,20 +92,33 @@ export default function CVItem({ cv }: CVCardProps) {
         setBase64Data(base64);
       };
 
-      getAuthorName();
       await getBlurCv();
-      revalidateImage();
-
-      const interval = setInterval(
-        revalidateImage,
-        Definitions.FETCH_WAIT_TIME * 1000
-      ); // Revalidate every 2 minutes
+      await getAuthorName();
+      await revalidateImage();
 
-      return () => clearInterval(interval);
+      interval.current = setInterval(() => {
+        revalidateImage();
+      }, Definitions.FETCH_WAIT_TIME * 1000); // Revalidate every 2 minutes
     };
 
     fetchData();
-  }, [cv, getBlur, getURL, getCachedUserName]);
+  }, [
+    cv.document_link,
+    cv.user_id,
+    getBlur,
+    getCachedUserName,
+    getURL,
+    revalidatePreview,
+  ]);
+
+  useEffect(
+    () => () => {
+      if (interval != null) {
```

Comment:
I believe if you'll do the check on `interval.current` ts won't require you to cast as Timeout on the next line